[BUG] [REGRESSION] Streaming responses regression

After upgrading from version 0.4.21 to 0.5.2, streaming responses no longer works with ollama
I have double checked by downgrading back to 0.4.21, and this is indeed a regression.
Steps to reproduce:
- Check the checkbox for `Enable Stream mode (Beta)`
- Attempt to send a prompt using the built in chat box in the plugin.

Expected result:
- The response should be streaming, and I should be able to stop the execution at any point (e.g. as soon as I get enough satisfactory information, which is why I use it)

Actual result:
- There is an error message pop up, and the following errors are shown in the `idea.log` file

```
2025-03-27 11:06:17,111 [  22592]   INFO - STDOUT - 11:06:17.110 [prompt-exec-1] ERROR c.d.g.s.p.s.StreamingPromptStrategy - Error in streaming prompt execution
2025-03-27 11:06:17,111 [  22592]   INFO - STDOUT - java.lang.IllegalArgumentException: messages cannot be null or empty
2025-03-27 11:06:17,111 [  22592]   INFO - STDOUT - 	at dev.langchain4j.internal.Exceptions.illegalArgument(Exceptions.java:19)
2025-03-27 11:06:17,111 [  22592]   INFO - STDOUT - 	at dev.langchain4j.internal.ValidationUtils.ensureNotEmpty(ValidationUtils.java:65)
2025-03-27 11:06:17,111 [  22592]   INFO - STDOUT - 	at dev.langchain4j.model.chat.request.ChatRequest.<init>(ChatRequest.java:22)
2025-03-27 11:06:17,111 [  22592]   INFO - STDOUT - 	at dev.langchain4j.model.chat.request.ChatRequest$Builder.build(ChatRequest.java:149)
2025-03-27 11:06:17,111 [  22592]   INFO - STDOUT - 	at dev.langchain4j.model.chat.StreamingChatLanguageModel.chat(StreamingChatLanguageModel.java:97)
2025-03-27 11:06:17,111 [  22592]   INFO - STDOUT - 	at com.devoxx.genie.service.prompt.strategy.StreamingPromptStrategy.lambda$executeStrategySpecific$2(StreamingPromptStrategy.java:95)
2025-03-27 11:06:17,111 [  22592]   INFO - STDOUT - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
2025-03-27 11:06:17,111 [  22592]   INFO - STDOUT - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
2025-03-27 11:06:17,111 [  22592]   INFO - STDOUT - 	at java.base/java.lang.Thread.run(Thread.java:1583)
2025-03-27 11:06:17,111 [  22592]   INFO - STDOUT - 11:06:17.111 [prompt-exec-1] ERROR c.d.g.s.prompt.error.PromptException - WARNING:Error during streaming response - false
2025-03-27 11:06:17,112 [  22593]   INFO - STDOUT - 11:06:17.112 [prompt-exec-1] WARN  c.d.g.s.p.error.PromptErrorHandler - Error during streaming response
2025-03-27 11:06:17,112 [  22593]   INFO - STDOUT - java.lang.IllegalArgumentException: messages cannot be null or empty
2025-03-27 11:06:17,112 [  22593]   INFO - STDOUT - 	at dev.langchain4j.internal.Exceptions.illegalArgument(Exceptions.java:19)
2025-03-27 11:06:17,112 [  22593]   INFO - STDOUT - 	at dev.langchain4j.internal.ValidationUtils.ensureNotEmpty(ValidationUtils.java:65)
2025-03-27 11:06:17,112 [  22593]   INFO - STDOUT - 	at dev.langchain4j.model.chat.request.ChatRequest.<init>(ChatRequest.java:22)
2025-03-27 11:06:17,112 [  22593]   INFO - STDOUT - 	at dev.langchain4j.model.chat.request.ChatRequest$Builder.build(ChatRequest.java:149)
2025-03-27 11:06:17,112 [  22593]   INFO - STDOUT - 	at dev.langchain4j.model.chat.StreamingChatLanguageModel.chat(StreamingChatLanguageModel.java:97)
2025-03-27 11:06:17,112 [  22593]   INFO - STDOUT - 	at com.devoxx.genie.service.prompt.strategy.StreamingPromptStrategy.lambda$executeStrategySpecific$2(StreamingPromptStrategy.java:95)
2025-03-27 11:06:17,112 [  22593]   INFO - STDOUT - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
2025-03-27 11:06:17,112 [  22593]   INFO - STDOUT - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
2025-03-27 11:06:17,112 [  22593]   INFO - STDOUT - 	at java.base/java.lang.Thread.run(Thread.java:1583)
2025-03-27 11:06:17,113 [  22594]   INFO - STDOUT - 11:06:17.113 [prompt-exec-1] ERROR c.d.g.s.p.s.StreamingResponseHandler - Streaming error for context 1743087977054: messages cannot be null or empty
2025-03-27 11:06:17,113 [  22594]   INFO - STDOUT - 11:06:17.113 [prompt-exec-1] ERROR c.d.g.s.p.s.StreamingPromptStrategy - Streaming error for context 1743087977054: Error during streaming response
2025-03-27 11:06:17,114 [  22595]   INFO - STDOUT - 11:06:17.114 [prompt-exec-1] DEBUG c.d.g.s.p.t.PromptTaskTracker - Task 17bc1138-2547143014766 with context ID 1743087977054 completed for project 17bc1138
2025-03-27 11:06:17,114 [  22595]   INFO - STDOUT - 11:06:17.114 [prompt-exec-1] DEBUG c.d.g.s.prompt.threading.PromptTask - Task 17bc1138-2547143014766 completed exceptionally: Error during streaming response
2025-03-27 11:06:17,217 [  22698]   INFO - STDOUT - 11:06:17.216 [AWT-EventQueue-0] ERROR c.d.g.s.prompt.error.PromptException - ERROR:Error occurred during prompt execution - true
2025-03-27 11:06:17,217 [  22698]   INFO - STDOUT - 11:06:17.217 [AWT-EventQueue-0] ERROR c.d.g.s.p.error.PromptErrorHandler - Error occurred during prompt execution
2025-03-27 11:06:17,217 [  22698]   INFO - STDOUT - com.devoxx.genie.service.prompt.error.StreamingException: Error during streaming response
2025-03-27 11:06:17,217 [  22698]   INFO - STDOUT - 	at com.devoxx.genie.service.prompt.streaming.StreamingResponseHandler.onError(StreamingResponseHandler.java:109)
2025-03-27 11:06:17,217 [  22698]   INFO - STDOUT - 	at com.devoxx.genie.service.prompt.strategy.StreamingPromptStrategy.lambda$executeStrategySpecific$2(StreamingPromptStrategy.java:98)
2025-03-27 11:06:17,217 [  22698]   INFO - STDOUT - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
2025-03-27 11:06:17,217 [  22698]   INFO - STDOUT - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
2025-03-27 11:06:17,217 [  22698]   INFO - STDOUT - 	at java.base/java.lang.Thread.run(Thread.java:1583)
2025-03-27 11:06:17,217 [  22698]   INFO - STDOUT - Caused by: java.lang.IllegalArgumentException: messages cannot be null or empty
2025-03-27 11:06:17,217 [  22698]   INFO - STDOUT - 	at dev.langchain4j.internal.Exceptions.illegalArgument(Exceptions.java:19)
2025-03-27 11:06:17,217 [  22698]   INFO - STDOUT - 	at dev.langchain4j.internal.ValidationUtils.ensureNotEmpty(ValidationUtils.java:65)
2025-03-27 11:06:17,217 [  22698]   INFO - STDOUT - 	at dev.langchain4j.model.chat.request.ChatRequest.<init>(ChatRequest.java:22)
2025-03-27 11:06:17,217 [  22698]   INFO - STDOUT - 	at dev.langchain4j.model.chat.request.ChatRequest$Builder.build(ChatRequest.java:149)
2025-03-27 11:06:17,217 [  22698]   INFO - STDOUT - 	at dev.langchain4j.model.chat.StreamingChatLanguageModel.chat(StreamingChatLanguageModel.java:97)
2025-03-27 11:06:17,217 [  22698]   INFO - STDOUT - 	at com.devoxx.genie.service.prompt.strategy.StreamingPromptStrategy.lambda$executeStrategySpecific$2(StreamingPromptStrategy.java:95)
2025-03-27 11:06:17,217 [  22698]   INFO - STDOUT - 	... 3 common frames omitted
2025-03-27 11:06:17,232 [  22713]   INFO - STDOUT - 11:06:17.232 [AWT-EventQueue-0] DEBUG c.d.g.s.p.c.PromptCancellationService - Unregistered execution: project=17bc1138, contextId=1743087977054
```

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -7,7 +7,7 @@ plugins {
 }
 
 group = "com.devoxx.genie"
-version = "0.5.2"
+version = "0.5.3"
 
 repositories {
     mavenCentral()

--- a/src/main/java/com/devoxx/genie/service/prompt/memory/ChatMemoryManager.java
+++ b/src/main/java/com/devoxx/genie/service/prompt/memory/ChatMemoryManager.java
@@ -140,6 +140,29 @@ public class ChatMemoryManager {
     }
 
     /**
+     * Adds user message to memory from the provided context
+     * @param context The chat message context containing the user message
+     */
+    public void addUserMessage(@NotNull ChatMessageContext context) {
+        try {
+            if (context.getUserMessage() == null && context.getUserPrompt() != null) {
+                // Create user message if not already set
+                context.setUserMessage(UserMessage.from(context.getUserPrompt()));
+            }
+            
+            if (context.getUserMessage() != null) {
+                log.debug("Adding user message to memory for context ID: {}", context.getId());
+                chatMemoryService.addMessage(context.getProject(), context.getUserMessage());
+                log.debug("Successfully added user message to memory");
+            } else {
+                log.warn("Attempted to add null user message to memory for context ID: {}", context.getId());
+            }
+        } catch (Exception e) {
+            throw new MemoryException("Failed to add user message to memory", e);
+        }
+    }
+
+    /**
      * Removes only the last AI message from memory
      * @param context The chat message context containing the AI message to remove
      */

--- a/src/main/java/com/devoxx/genie/service/prompt/strategy/AbstractPromptExecutionStrategy.java
+++ b/src/main/java/com/devoxx/genie/service/prompt/strategy/AbstractPromptExecutionStrategy.java
@@ -80,15 +80,14 @@ public abstract class AbstractPromptExecutionStrategy implements PromptExecution
      */
     protected abstract String getStrategyName();
     
-//    /**
-//     * Prepares memory with system message if needed and adds the user message.
-//     *
-//     * @param context The chat message context
-//     */
-//    protected void prepareMemory(@NotNull ChatMessageContext context) {
-//        chatMemoryManager.prepareMemory(context);
-//        // chatMemoryManager.addUserMessage(context);
-//    }
+    /**
+     * Prepares memory with system message if needed and adds the user message.
+     *
+     * @param context The chat message context
+     */
+    protected void prepareMemory(@NotNull ChatMessageContext context) {
+        chatMemoryManager.prepareMemory(context);
+    }
     
     /**
      * Standardized error handling for execution exceptions.

--- a/src/main/java/com/devoxx/genie/service/prompt/strategy/NonStreamingPromptStrategy.java
+++ b/src/main/java/com/devoxx/genie/service/prompt/strategy/NonStreamingPromptStrategy.java
@@ -59,6 +59,10 @@ public class NonStreamingPromptStrategy extends AbstractPromptExecutionStrategy 
         log.debug("Before memory preparation - context ID: {}", context.getId());
         chatMemoryManager.logMemoryState(context.getProject());
         
+        // Prepare memory and add user message
+        prepareMemory(context);
+        chatMemoryManager.addUserMessage(context);
+        
         log.debug("After memory preparation - context ID: {}", context.getId());
         chatMemoryManager.logMemoryState(context.getProject());
 

--- a/src/main/java/com/devoxx/genie/service/prompt/strategy/StreamingPromptStrategy.java
+++ b/src/main/java/com/devoxx/genie/service/prompt/strategy/StreamingPromptStrategy.java
@@ -55,7 +55,9 @@ public class StreamingPromptStrategy extends AbstractPromptExecutionStrategy {
         log.debug("Before memory preparation (streaming) - context ID: {}", context.getId());
         chatMemoryManager.logMemoryState(context.getProject());
         
-        // prepareMemory(context);
+        // Prepare memory and add user message
+        prepareMemory(context);
+        chatMemoryManager.addUserMessage(context);
         
         log.debug("After memory preparation (streaming) - context ID: {}", context.getId());
         chatMemoryManager.logMemoryState(context.getProject());

--- a/src/main/resources/META-INF/plugin.xml
+++ b/src/main/resources/META-INF/plugin.xml
@@ -38,6 +38,7 @@
         <h2>v0.5.3</h2>
         <UL>
             <Li>Feat #586 : Added Gemini 2.5 Pro model by @stephanj</LI>
+            <LI>Fix #588 : Fix streaming responses regression by @stephanj</LI>
         </UL>
         <h2>v0.5.2</h2>
         <UL>

--- a/src/main/resources/application.properties
+++ b/src/main/resources/application.properties
@@ -1,1 +1,1 @@
-version=0.5.2
+version=0.5.3

